renpy/revertable.py

```diff
@@ -411,10 +411,54 @@ def _rollback(self, compressed):
         set.update(self, compressed)
 
 
-class RevertableObject(object):
+class RevertableObjectMeta(type):
+    def __new__(cls, name, bases, dct):
+        rv = super(RevertableObjectMeta, cls).__new__(cls, name, bases, dct)
+
+        # Collect all slots from type MRO
+        slots = set()
+        for t in rv.__mro__:
+            # Stop on object as it has no slots but it is special
+            if t is object:
+                break
+
+            if "__slots__" in t.__dict__:
+                slots.update(t.__slots__)
```

Comment:
This misbehaves if a subclass has a string \_\_slots__.
Something like `slots.update((t.__slots__,) if isinstance(t.__slots__, str) else t.__slots__)`.